File: TESTS/test-01/armstrong.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool check_armstrong (int number , int size)
{
    /* 
        Armstrong number is a number in which a digit in a number 
        is is raised to the power of no of digits is called armstong number 
    */
   int temp = number;
   if (number == 0)
       return true;

    if (number < 0)
       return false;

    int last_digit = 0;
    int multipliyer = size;
    int result = 0;
    while (number > 0)
    {
        last_digit = number % 10; // taking last digit 
        number = number / 10; // removing last digit

        result += pow(last_digit,size);
    }

    if (temp == result)
    {
        return true;
    }

    return false;
}

int main ()
{
    int number;
    cout<<"Please enter the number "<<endl;
    cin>>number;

    int size = (number == 0) ? 1 : (int) log10(number) + 1; // this automatically calculates the size of the number
    int value = check_armstrong(number,size);

    // checking if the number is armstrong
    if(value)
    {
        cout<<"Yes the number is armstrong";
    }
    else
    {
        cout<<"No the number is not armstrong";
    }
    return 0;
}

File: python/basic/Armstrong.py
def power(x,y):
    if(y==0):
        return 1
    half = power(x,y//2)

    if ( y % 2 == 0):
        result = half * half 
        return result 
    else:
        result = half * half 
        result = x * result
        return result 

def sizeof(x):
    temp = x
    count = 0
    while (temp > 0):
        temp //=10
        count += 1
    return count 

def armstrong(x):
    size = sizeof(x)
    temp = x
    sum = 0 
    while (temp > 0):
        last = temp % 10
        sum += power(last,size)
        temp //=10
    if(sum == x):
        return 1 
    else:
        return 0

x = int(input("Please enter the value that you want to check if:\n"))
flag = armstrong(x)
if (flag == 1):
    print("yes it is armstrong number")
else :
    print("No it it not armstrong number")